dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunction.cs

```diff
@@ -432,7 +433,12 @@ await using (enumerator.ConfigureAwait(false))
         // visible to a consumer if that's needed.
         if (ex is OperationCanceledException cancelEx)
         {
-            throw new KernelFunctionCanceledException(kernel, kernelFunction, arguments, result, cancelEx);
+            var kernelEx = new KernelFunctionCanceledException(kernel, kernelFunction, arguments, result, cancelEx);
+            foreach (DictionaryEntry entry in cancelEx.Data)
+            {
+                kernelEx.Data.Add(entry.Key, entry.Value);
```

Comment:
Is it worth to add a unit test to verify this logic, just in case? :) Since it's `KernelFunction` class, this place is kind of sensitive.

Reply:
Even if we know it works fine, having a unit test is nice to avoid missing regressions in the future 😄